Machine subclasses

rbns opened the pull request:
Use subclasses of Machine for virtual machines and switches

Commit:
cherry pick the schema changes

Commit:
split Machine model into Machine, VirtualMachine and Switch

Commit:
fix machine update service specs

Commit:
cleanup

Commit:
cleanup

Commit:
switch ports now belong to switch class

Commit:
fix history for machine subclasses.

rbns committed:
Merge branch 'develop' into machine-subclasses

dra reviewed (commented):
Imho the EditableMachineForm should not rely on String-matching to differ between the types, but there should be constants used throughout the code to differ the device types.
And the type-selector should sent a number or constant String to match the device type; in case one wants to edit the displayed value for one of the types this has to be done on various locations.

Commit:
fix hardcoded lists of machine types

the list of machine types is now generated by inspecting the subclasses of Machine, calling their device_type_name method to get a name string. this name string can be used in the device_type_by_name method of Machine to get the matching class to a string.

This removes the need to have an additional list / constants for the different types. Inheriting from Machine and implementing the device_type_name method adds a class to the list.
